src/builder/unzip.rs

```diff
@@ -0,0 +1,70 @@
+use std::fs::File;
+use std::path::{Path, PathBuf};
+
+use zip::ZipArchive;
+
+use config::builders::ZipInfo;
+use super::context::Context;
+use super::download::download_file;
+use file_util::{copy_stream, create_dir};
+use path_util::ToRelative;
+
+
+pub fn unzip_file(_ctx: &mut Context, src: &Path, dst: &Path, subdir: &Path)
+    -> Result<(), String>
+{
+    let file = try_msg!(File::open(src),
+        "Cannot open archive: {err}");
+    let mut zip = try_msg!(ZipArchive::new(file),
+        "Error when unpacking zip archive: {err}");
+
+    // TODO check sha256 sum
```

Comment:
Well I you add an option, please implement it. It's absent for Tar command is just because it was far far ago, when we didn't have cargo and so a good library that implements this thing. Now it's easy to implement.